notebook/services/contents/tests/test_contents_api.py

```diff
@@ -522,6 +522,12 @@ def test_delete_dirs(self):
         listing = self.api.list('/').json()['content']
         self.assertEqual(listing, [])
 
+    def test_delete_non_empty_dir(self):
+        # Test that non empty directory can be deleted
+        self.api.delete(u'å b')
+        # Assertion will pass only if self.api.delete does not throw and error
```

Comment:
A test doesn't have to have an assert - you can leave this off, and it will still fail if there's an error. When a test just checks that something runs without an error, that's a 'smoketest' (as in 'does this make smoke come out').

However, it would be good to check that it has actually been removed as well. Maybe copy some code from `test_list_nonexistant_dir`.

Reply:
Oh okay. Sure, I'll make these changes!